### ISSUE TITLE
An internal server error occurred

### ISSUE THREAD

**Issue by westy92** [ngonzalvez/rest-facade#22]
Our team is still trying to determine the root cause of this error, but we know a few things:

rest-facade version is 1.7.1.

Stack trace:
```
Internal Server Error: An internal server error occurred
    at /usr/src/node_modules/rest-facade/src/Client.js:321:22
    at Request.callback (/usr/src/node_modules/superagent/lib/node/index.js:699:3)
    at /usr/src/node_modules/superagent/lib/node/index.js:868:18
    at Stream.<anonymous> (/usr/src/node_modules/superagent/lib/node/parsers/json.js:16:7)
    at emitNone (events.js:105:13)
    at Stream.emit (events.js:207:7)
    at Unzip.<anonymous> (/usr/src/node_modules/superagent/lib/node/unzip.js:53:12)
    at emitNone (events.js:110:20)
    at Unzip.emit (events.js:207:7)
    at endReadableNT (_stream_readable.js:1059:12)
    at _combinedTickCallback (internal/process/next_tick.js:138:11)
    at process._tickDomainCallback (internal/process/next_tick.js:218:9)
```

Usage: from a high level we are using the `auth0` npm package (which depends on rest-facade).  Auth0 is having intermittent isses with one of their endpoints (https://status.auth0.com/incidents/71nlbr16j7ng).  It appears this error is now getting trapped inside rest-facade.

**Comment by westy92** [comment]
A team member is getting the same error with rest-facade v1.5.

```
error: Internal Server Error. stack=Internal Server Error: An internal server error occurred
    at /home/anthony/dev/api/node_modules/rest-facade/src/Client.js:293:22
    at Request.callback (/home/anthony/dev/api/node_modules/superagent/lib/node/index.js:688:3)
    at /home/anthony/dev/api/node_modules/superagent/lib/node/index.js:883:18
    at Stream.<anonymous> (/home/anthony/dev/api/node_modules/superagent/lib/node/parsers/json.js:16:7)
    at emitNone (events.js:86:13)
    at Stream.emit (events.js:185:7)
    at Unzip.<anonymous> (/home/anthony/dev/api/node_modules/superagent/lib/node/unzip.js:53:12)
    at emitNone (events.js:91:20)
    at Unzip.emit (events.js:185:7)
    at endReadableNT (_stream_readable.js:974:12)
    at _combinedTickCallback (internal/process/next_tick.js:80:11)
    at process._tickDomainCallback (internal/process/next_tick.js:128:9)
```

**Comment by ngonzalvez** [comment]
Hi @westy92, sorry for the delay. 

Every time an error occurs rest-facade constructs an APIError using the error returned by the API it is consuming. However it didn't work for some specific cases. I just merged PR #23 (thanks @iamjochem) which should give you more meaningful error messages. The new version is 1.7.2, try it and let me know if you have any further issues.

Again, sorry for taking so long to reply.

**Comment by ngonzalvez** [comment]
I am closing this issue. We'll reopen it if anyone reports this issue again.
